### PR TITLE
Add 7.0, 7.1.1, 8.0, 8.1.1 to unmarshalling CustomNavigationView.SavedState hack

### LINKED ISSUE #1400: Unmarshalling: NavigationView$SavedState crash in drawer
```
Caused by android.os.BadParcelableException: ClassNotFoundException when unmarshalling: 
android.support.design.widget.NavigationView$SavedState.ui.views.drawer.CustomNavigationView$SavedState.<init>(CustomNavigationView.java:127)
```

### PR DESCRIPTION
**Pull request by EmmanuelMess**
Fixes #1400

### CODE REVIEW

File: app/src/main/java/com/amaze/filemanager/ui/views/drawer/CustomNavigationView.java [hunk 1]
```diff
@@ -105,15 +105,20 @@ public void onRestoreInstanceState(Parcelable state) {
...
      * I could be wrong.
      * For the crash see https://github.com/TeamAmaze/AmazeFileManager/issues/1101.
      */
     public boolean doesNavigationViewSavedStateExist() {
-        return Build.VERSION.SDK_INT != Build.VERSION_CODES.M;
+        return Build.VERSION.SDK_INT != Build.VERSION_CODES.M
+                && Build.VERSION.SDK_INT != Build.VERSION_CODES.N
+                && Build.VERSION.SDK_INT != Build.VERSION_CODES.N_MR1
+                && Build.VERSION.SDK_INT != Build.VERSION_CODES.O
+                && Build.VERSION.SDK_INT != Build.VERSION_CODES.O_MR1;
```
Comment (TranceLove) [inline app/src/main/java/com/amaze/filemanager/ui/views/drawer/CustomNavigationView.java]: If we're certain `Build.VERSION_CODES` is a series of consequential integers without skipping, why not just do
```
return Build.VERSION.SDK_INT >= Build.VERSION_CODES.M
     && Build.VERSION.SDK_INT <= Build.VERSION_CODES.O_MR1;
```

Comment (VishalNehra) [reply app/src/main/java/com/amaze/filemanager/ui/views/drawer/CustomNavigationView.java]: More like ````return Build.VERSION.SDK_INT<Build.VERSION_CODES.M````

Comment (TranceLove) [reply app/src/main/java/com/amaze/filemanager/ui/views/drawer/CustomNavigationView.java]: Right, **is not** between Android M and Android O MR1.

Comment (EmmanuelMess) [reply app/src/main/java/com/amaze/filemanager/ui/views/drawer/CustomNavigationView.java]: > More like `return Build.VERSION.SDK_INT<Build.VERSION_CODES.M`

Remember that we don't know if future versions will have this issue.

Anyway, fixed.

Comment (VishalNehra) [reply app/src/main/java/com/amaze/filemanager/ui/views/drawer/CustomNavigationView.java]: @EmmanuelMess I think you directly picked @TranceLove 's code without negating it. Please check once.

**Comment by EmmanuelMess** [comment]
@VishalNehra ?

**Comment by VishalNehra** [comment]
@EmmanuelMess I have a pending review here.

**Comment by EmmanuelMess** [comment]
@VishalNehra 

screenshot_2018-10-28 add 7 0 7 1 1 8 0 8 1 1 to unmarshalling customnavigationview savedstate hack by emmanuelmess pu

File: app/src/main/java/com/amaze/filemanager/ui/views/drawer/CustomNavigationView.java [hunk 2]
```diff
@@ -67,7 +66,7 @@ public void deselectItems() {
 
     @Override
     public Parcelable onSaveInstanceState() {
-        if (!doesNavigationViewSavedStateExist()) {
+        if (!isNavigationViewSavedStateMissing()) {
```
[the remark this answers is not in the document]
Comment (EmmanuelMess) [reply app/src/main/java/com/amaze/filemanager/ui/views/drawer/CustomNavigationView.java]: Fixed.

**Pull request merged**